ledger/accountdb.go

```diff
@@ -633,6 +644,406 @@ func accountsAddNormalizedBalance(tx *sql.Tx, proto config.ConsensusParams) erro
 	return rows.Err()
 }
 
+// accountsCreateResourceTable creates the resource table in the database.
+func accountsCreateResourceTable(ctx context.Context, tx *sql.Tx) error {
+	var exists bool
+	err := tx.QueryRowContext(ctx, "SELECT 1 FROM pragma_table_info('resources') WHERE name='addrid'").Scan(&exists)
+	if err == nil {
+		// Already exists.
+		return nil
+	}
+	if err != sql.ErrNoRows {
+		return err
+	}
+	for _, stmt := range createResourcesTable {
+		_, err = tx.ExecContext(ctx, stmt)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type baseOnlineAccountData struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	VoteID          crypto.OneTimeSignatureVerifier `codec:"A"`
```

Comment:
I understand it does not matter but... maybe "v", "s", "f", "l", "d" for a little bit more semantics?

Reply:
this is an embedded struct, right ? we need to make sure that it's not overlapping with the codec mapping.
I was thinking that just picking the alphabet would make it easier to avoid "searching for a gap" in the future.

Reply:
yeah, very similar to protobuf, which just uses numbers

Reply:
btw - the reason I use the `baseOnlineAccountData` here is because it's going to be moving to a different table later on. I figured that it would be easier to have it already defined separately.